src/python_testing/TC_SUTestBase.py

```diff
@@ -0,0 +1,480 @@
+                dir=storage_dir, prefix='provider_', suffix='.log', delete=True, mode='ab')
+            logger.info(f"Writing Provider logs at :{log_file.name}")
+        else:
+            logger.info(f"Writing Provider logs at : {log_file}")
+        # Launch the Provider subprocess using the Wrapper
+        proc = OTAProviderSubprocess(
+            self.provider_app_path,
+            storage_dir=storage_dir,
+            port=port,
+            discriminator=discriminator,
+            passcode=setup_pincode,
+            ota_source=ota_image_path,
+            extra_args=extra_args,
+            kvs_path=kvs_path,
+            log_file=log_file,
+            err_log_file=log_file)
+        proc.start(
+            expected_output=expected_output,
+            timeout=timeout)
+
+        self.current_provider_app_proc = proc
+        logger.info(f"Provider started with PID:  {self.current_provider_app_proc.get_pid()}")
+
+    async def announce_ota_provider(self,
+                                    controller: ChipDeviceCtrl,
+                                    provider_node_id: int,
+                                    requestor_node_id: int,
+                                    reason: Clusters.OtaSoftwareUpdateRequestor.Enums.AnnouncementReasonEnum = Clusters.OtaSoftwareUpdateRequestor.Enums.AnnouncementReasonEnum.kUpdateAvailable,
+                                    vendor_id: int = 0xFFF1,
+                                    endpoint: int = 0):
+        """Announce the OTA provider that a software update is requested.
+
+        Args:
+            controller (ChipDeviceCtrl): Controller for DUT
+            provider_node_id (int): Node id for the provider
+            requestor_node_id (int): Node id for the requestor
+            reason (Clusters.OtaSoftwareUpdateRequestor.Enums.AnnouncementReasonEnum, optional): Update Reason. Defaults to Clusters.OtaSoftwareUpdateRequestor.Enums.AnnouncementReasonEnum.kUpdateAvailable.
+            vendor_id (int, optional): Vendor id. Defaults to 0xFFF1.
+            endpoint (int, optional): Endpoint id. Defaults to 0.
+
+        Returns:
+            _type_: _description_
+        """
+        cmd_announce_ota_provider = Clusters.OtaSoftwareUpdateRequestor.Commands.AnnounceOTAProvider(
+            providerNodeID=provider_node_id,
+            vendorID=vendor_id,
+            announcementReason=reason,
+            metadataForNode=None,
+            endpoint=endpoint
+        )
+        logger.info("Sending AnnounceOTA Provider Command")
+        cmd_resp = await self.send_single_cmd(
+            cmd=cmd_announce_ota_provider,
+            dev_ctrl=controller,
+            node_id=requestor_node_id,
+            endpoint=endpoint,
+        )
+        logger.info(f"Announce command sent {cmd_resp}")
+        return cmd_resp
+
+    async def write_providers_into_requestor(self, controller: ChipDeviceCtrl, provider_node_id: int, requestor_node_id: int, endpoint: int = 0):
```

Comment:
nit: I would rename it to `set_default_ota_providers_list` for clarity, mentioning in the docstring that you are writing attribute on requestor